### ISSUE TITLE
Support starship for shell customization

### ISSUE THREAD

**Issue by marcelotrevisani-centrica** [prefix-dev/shell#132]
It would be really nice if it could support some shell interface customizations or support starship out of the box
https://starship.rs/

**Comment by certik** [comment]
Yes, we'll have to figure out how to do shell customizations. The only requirement is that whatever default we use, it must be fast (instant). Other than that, we should use great defaults, and then allow users to customize more as needed.
